tracer/src/Datadog.Trace/DuckTyping/DuckType.cs

```diff
@@ -921,17 +1045,17 @@ private static Delegate CreateStructCopyMethod(ModuleBuilder moduleBuilder, Type
             /// <param name="targetType">Target type</param>
             /// <param name="activator">Proxy activator</param>
             /// <param name="exceptionInfo">Exception dispatch info instance</param>
-            internal CreateTypeResult(Type proxyTypeDefinition, Type proxyType, Type targetType, Delegate activator, ExceptionDispatchInfo exceptionInfo)
+            internal CreateTypeResult(Type proxyTypeDefinition, Type? proxyType, Type targetType, Delegate? activator, ExceptionDispatchInfo? exceptionInfo)
             {
-                TargetType = targetType;
-                _proxyType = proxyType;
                 _activator = activator;
+                _proxyType = proxyType;
                 _exceptionInfo = exceptionInfo;
+                TargetType = targetType;
                 Success = proxyType != null && exceptionInfo == null;
-                if (exceptionInfo != null)
+                if (exceptionInfo is not null)
                 {
-                    MethodInfo methodInfo = typeof(CreateTypeResult).GetMethod(nameof(ThrowOnError), BindingFlags.NonPublic | BindingFlags.Instance);
-                    _activator = methodInfo
+                    MethodInfo? methodInfo = typeof(CreateTypeResult).GetMethod(nameof(ThrowOnError), BindingFlags.NonPublic | BindingFlags.Instance);
+                    _activator = methodInfo!
```

Comment:
```suggestion
                    MethodInfo methodInfo = typeof(CreateTypeResult).GetMethod(nameof(ThrowOnError), BindingFlags.NonPublic | BindingFlags.Instance)!;
                    _activator = methodInfo
```

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/2716/commits/7a8a8942d10547d8c3d564471e01f9631f2c635d